### ISSUE TITLE
Bug: list_modify and zap do not remove key-value pairs whose values are lists

### ISSUE THREAD

**Issue by adamroyjones** [tidyverse/purrr#776]
I am using purrr 0.3.4 with R 3.6.3 (on Ubuntu 20.04).

The [documentation](https://purrr.tidyverse.org/reference/list_modify.html) to `list_modify` indicates that one should "[u]se `zap()` to remove values" from a list. This doesn't seem to work in the way that I, a naïve user, would expect: it does not work when the value associated with a key is a list:

```r
library(purrr)

list(foo = 1, bar = 2) %>% list_modify(foo = zap())
# { bar: 2 }

list(foo = c(1, 2), bar = 3) %>% list_modify(foo = zap())
# { bar: 3 }

list(foo = list(fst = 1, snd = 2), bar = 3) %>% list_modify(foo = zap())
# { foo: { fst: 1, snd: 2 }, bar: 3 }
```

(This behaviour is reflected in the documentation itself.)

Is this expected behaviour? Am I being an idiot?

**Comment by adamroyjones** [comment]
I should add that, if this turns out to be unexpected behaviour, I'm happy to have a stab at fixing it.

**Comment by adamroyjones** [comment]
I've raised a pull request that hopefully fixes this (#777).
